libreyolo/models/rfdetr/model.py

```
@@ -927,11 +994,68 @@ def _load_semantic_weights(self, model_path: str | dict[str, Any]) -> None:
+
+        ckpt_nc = loaded.get("nc")
+        if ckpt_nc is None:
+            names = loaded.get("names")
+            ckpt_nc = len(names) if names else None
+        if ckpt_nc is None:
+            state = _checkpoint_model_state(loaded)
+            ckpt_nc = 1 if state.get("depth_head.weight") is not None else None
+        if ckpt_nc is not None and int(ckpt_nc) != 1:
+            raise RuntimeError(
+                f"RF-DETR depth checkpoints must declare nc=1, got nc={ckpt_nc}."
+            )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)